### DISCUSSION TITLE
How to modify collection as result of property change

### DISCUSSION THREAD

**Discussion by ngm011** [MarimerLLC/csla#3228] [Questions] [answered]
Hi there,

Considering that the only way to modify property value as a result of some other property value change is to use a rule initialized with affected property and call AddOutValue on the context for a given affected/dependent property.

Keeping that in mind, how to change the collection exposed by dependent property so that items can be added or removed once the primary property is changed without changing the instance of the collection?

In other words, if I had a property named "Type", every time its value is changed I want to add or remove some items of the child collection exposed by property named "Cells".

Thanks,

--- ngm

**Comment by Chicagoan2016** [comment, marked as answer]
Both 'Type' and child collection properties should be 'sent' to the business rule.
I think in this case you have to use the 'Target' inside the business rule.

**Reply by rockfordlhotka** [reply]
I agree with @Chicagoan2016 - this is a case where the rules engine doesn't have the capability to do what you want by just adding out values - you need to use `context.Target` to directly manipulate the object graph.

The `Target` property gives you a reference to the business object against which the rule is running: the target object. You can use that reference to navigate the entire object graph, including using the business object's `Parent` property if you need to go higher in the graph.

**Reply by ngm011** [reply]
Yeah, I figured. Just wanted to see if there's any other safer way around.

In other words, you'll be leaving "auto-pilot" mode so you are on your own here 😁 

Thx guys,

--- ngm
